packages/salesforcedx-utils-vscode/src/helpers/utils.ts

```diff
@@ -14,11 +14,13 @@ export function isNullOrUndefined(object: any): object is null | undefined {
 }
 
 export function extractJsonObject(str: string): any {
-
-  const jsonString = str.substring(
-    str.indexOf('{'),
-    str.lastIndexOf('}') + 1
-  );
+  const jsonString = str.substring(str.indexOf('{'), str.lastIndexOf('}') + 1);
 
   return JSON.parse(jsonString);
 }
+
```

Comment:
Should the type for predicate below be the same as the param for Array.map? `callbackfn: (value: T, index: number, array: T[]) => unknown`

Reply:
If that causes issues I'd think we probably need a code comment calling out how to use predicate

Reply:
done

Reply:
what did you end up doing for this?

Reply:
```
export async function asyncFilter<T>(arr: T[], callback: (value: T, index: number, array: T[]) => unknown) {
  const results = await Promise.all(arr.map(callback));

  return arr.filter((_v, index) => results[index]);
}
```